mdsobjects/cpp/mdstreeobjects.cpp

```diff
@@ -1661,7 +1681,337 @@ StringArray *TreeNodeArray::getUsage()
+{
+    this->tree = 0;
+    this->nid = nid;
+    this->connection = connection;
+    clazz = CLASS_S;
+    dtype = DTYPE_NID;
+    setAccessory(units, error, help, validation);
+}
+
+std::string  TreeNodeThinClient::getNciString(int itm)
+{
+    char expr[64];
+    sprintf(expr, "GETNCI(%d,%s)", nid, convertNciItm(itm));
+    Data *retStringData = connection->get(expr);
+    if(!retStringData)
+	throw MdsException("Error in Remote evaluation of getnci");
+    
+    AutoString as(retStringData->getString());
+//    char *retStringPtr = retStringData->getString();
+//    std::string retString(retStringPtr);
+//    delete [] retStringPtr;
+    MDSplus::deleteData(retStringData);
+    return as.string;
+}
+
+char TreeNodeThinClient::getNciChar(int itm)
+{
+    char expr[64];
+    sprintf(expr, "GETNCI(%d,%s)", nid, convertNciItm(itm));
+    Data *retData = connection->get(expr);
+     if(!retData)
+	throw MdsException("Error in Remote evaluation of getnci");
+    char retChar = retData->getByte();
+    MDSplus::deleteData(retData);
+    return retChar;
+}
+   
+int TreeNodeThinClient::getNciInt(int itm)
+{
+    char expr[64];
+    sprintf(expr, "GETNCI(%d,%s)", nid, convertNciItm(itm));
+    Data *retData = connection->get(expr);
+    if(!retData)
+	throw MdsException("Error in Remote evaluation of getnci");
+    int retInt = retData->getInt();
+    MDSplus::deleteData(retData);
+    return retInt;
+}
+
+int64_t TreeNodeThinClient::getNciInt64(int itm)
+{
+    char expr[64];
+    sprintf(expr, "GETNCI(%d,%s)", nid, convertNciItm(itm));
+    Data *retData = connection->get(expr);
+    if(!retData)
+	throw MdsException("Error in Remote evaluation of getnci");
+    int64_t retLong = retData->getLong();
+    MDSplus::deleteData(retData);
+    return retLong;
+}
+
+char *TreeNodeThinClient::getPath()
+{
+    char expr[64];
+    sprintf(expr, "GETNCI(%d,\'PATH\')", nid);
+    Data *retData = connection->get(expr);
+    if(!retData)
+	throw MdsException("Error in Remote evaluation of getnci(path)");
+    char *retPath = retData->getString();
+    MDSplus::deleteData(retData);
+    return retPath;
+}
+    
+EXPORT Data *TreeNodeThinClient::getData()
+{
+    char expr[64];
+    sprintf(expr, "GETNCI(%d,\'RECORD\')", nid);
+    Data *retData = connection->get(expr);
+    if(!retData)
+	throw MdsException("Error in Remote evaluation of getnci(record)");
+    return retData;
+}
+  
+EXPORT void TreeNodeThinClient::putData(Data *data)
+{
+    const char *path = (const char *)getPath();
+    Data *args[1];
+    args[0] = data->data();
+    connection->put(path, (char *)"$", args, 1);
+    MDSplus::deleteData(args[0]);
+    delete [] path;
+  
+}
+
+EXPORT void TreeNodeThinClient::deleteData()
+{
+    char *path = getFullPath();
+    Data *args[1];
+    args[0] = 0;
+    connection->put(path, (char *)"*", args, 0);
+    delete [] path;
+}
+
+EXPORT bool TreeNodeThinClient::isOn()
```

Comment:
Even in this case:

    AutoData<Data> d(connection->get(expr));
....etc.

    return d->getByte() == 0;  // This is really inverted logic?
}